### PR TITLE
Commonize `:compose:material3:adaptive` modules

### PR DESCRIPTION
**Pull request by terrakok**
Commonize and configure publishing for 
 - `:compose:material3:adaptive:adaptive`
 - `:compose:material3:adaptive:adaptive-layout`
 - `:compose:material3:adaptive:adaptive-navigation`
 - `:window:window-core`
modules.

Added new method to non-android source set to get `WindowAdaptiveInfo`. 

Fixes https://youtrack.jetbrains.com/issue/CMP-5817

## Testing
Checked it in the MPP demo app:
image

### Features - Multiple Platforms
- Commonized `:compose:material3:adaptive:adaptive`, `:compose:material3:adaptive:adaptive-layout`, `:compose:material3:adaptive:adaptive-navigation` and `:window:window-core` modules

### CODE REVIEW

**Comment by igordmn** [comment]
> Just copy-pasted material3/adaptive directory from the androidx to JetBrains fork

Please specify the source commit. is it `androidx/compose-material3/1.3.0-beta03`?

File: compose/material3/adaptive/adaptive-layout/src/jvmStubsMain/kotlin/androidx/compose/material3/adaptive/layout/NotImplemented.jvmStubs.kt
Comment (igordmn) [inline compose/material3/adaptive/adaptive-layout/src/jvmStubsMain/kotlin/androidx/compose/material3/adaptive/layout/NotImplemented.jvmStubs.kt]: This shouldn't be in `androidx/compose-material3/1.3.0-beta03`, the stubs still weren't merged to `integration` or `jb-main`

File: compose/material3/adaptive/adaptive-layout/src/commonTest/kotlin/androidx/compose/material3/adaptive/layout/PaneScaffoldDirectiveTest.kt
Comment (MatkovIvan) [inline compose/material3/adaptive/adaptive-layout/src/commonTest/kotlin/androidx/compose/material3/adaptive/layout/PaneScaffoldDirectiveTest.kt]: It's ported in `:kruth` module, no need to rewrite every check

Comment (terrakok) [reply compose/material3/adaptive/adaptive-layout/src/commonTest/kotlin/androidx/compose/material3/adaptive/layout/PaneScaffoldDirectiveTest.kt]: updated

File: MULTIPLATFORM.md
Comment (MatkovIvan) [inline MULTIPLATFORM.md:74]: Do we need to adopt our CI for it?

File: compose/material3/adaptive/adaptive-navigation/build.gradle
Comment (MatkovIvan) [inline compose/material3/adaptive/adaptive-navigation/build.gradle:1]: I'm surprised to see no dependencies to a navigation library here.

File: MULTIPLATFORM.md
Comment (terrakok) [reply MULTIPLATFORM.md:74]: No. By default we publish all libraries

File: compose/material3/adaptive/adaptive-navigation/build.gradle
Comment (terrakok) [reply compose/material3/adaptive/adaptive-navigation/build.gradle:1]: It is quite small module with independent logic how to behave when a screen has different layouts

File: libraryversions.toml
Comment (MatkovIvan) [inline libraryversions.toml:295]: "versions.WINDOW" is already `1.6.0-alpha01` in AOSP. Shouldn't we update it here?

File: gradle.properties
Comment (MatkovIvan) [inline gradle.properties]: The stable 1.3.0 was release a few months ago. Any reason to use alpha?

File: MULTIPLATFORM.md
Comment (MatkovIvan) [reply MULTIPLATFORM.md:74]: As far as I remember, we still need to specify a default version there. cc @igordmn

**Review by MatkovIvan — APPROVED** [review]
LGTM overall, no blockers. Please recheck the versions on our CI and redirecting artifacts before merge

File: gradle.properties
Comment (terrakok) [reply gradle.properties]: I didn't touch the version. As I understand the version was set in the commit we based our fork on

File: libraryversions.toml
Comment (terrakok) [reply libraryversions.toml:295]: I didn't touch the version. As I understand the version was set in the commit we based our fork on

File: gradle.properties
Comment (MatkovIvan) [reply gradle.properties]: window subfolder wasn't involved in specific version selection, If it's outdated, please merge the stable state (might be done separately)

Comment (MatkovIvan) [reply gradle.properties]: Actually, I'm almost sure that this version doesn't match to the commit/state of the code. Blocking the merge before explicit check

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by MatkovIvan — APPROVED** [review]

File: compose/material3/adaptive/adaptive-layout/src/commonTest/kotlin/androidx/compose/material3/adaptive/layout/ThreePaneMotionTest.kt
Comment (igordmn) [inline compose/material3/adaptive/adaptive-layout/src/commonTest/kotlin/androidx/compose/material3/adaptive/layout/ThreePaneMotionTest.kt]: Let's reset formatting to avoid merge conflicts (in the other files too)

File: gradle.properties
Comment (igordmn) [inline gradle.properties]: We didn't take into account that material3 adaptive has own release cycle.

Window-core has a separate release cycle as well.

Because of that we need:
- find a commit for the latest version
- partially merge subfolders of this commit to jb-main.

Please, create tasks on me:
- merge the latest material3.adaptive
- merge the latest androidx.window.core
- enable publication of material3.adaptive/androidx.window.core

Until we merge them, these specified versions can be unaligned with the actual state of the source, and we can't publish.

Let's remove also changes in `mpp/build.gradle.kts`, then we can merge the PR in the current state.

File: libraryversions.toml
Comment (igordmn) [reply libraryversions.toml:295]: This is because we don't merge `window` and `libversions` to jb-main, they can be old. The actual state is in `integration`, but as with any other lib we merge only released commits to `jb-main`

File: MULTIPLATFORM.md
Comment (igordmn) [reply MULTIPLATFORM.md:74]: Yes, we need, but I'll done it myself before the release, see my other comment, we can't enable publication just now

**Review by igordmn — CHANGES_REQUESTED** [review]

File: gradle.properties
Comment (igordmn) [reply gradle.properties]: >  then we can merge the PR in the current state.

Actually it might be better to merge it after, but I am not sure when I able to merge these folders.

File: gradle.properties
Comment (igordmn) [inline gradle.properties]:
```suggestion
artifactRedirecting.androidx.compose.material3.adaptive.version=1.0.0-beta04
```

File: gradle.properties
Comment (igordmn) [inline gradle.properties]:
```suggestion
artifactRedirecting.androidx.window.core.version=1.3.0
```

**Review by igordmn — APPROVED** [review]

**Pull request merged**
